portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,22 +24,35 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/literature")
+/** Servlet that maintains the comment section.*/
+@WebServlet("/comment")
 public final class DataServlet extends HttpServlet {
 
-  private List<String> BookRecs;
+  private CommentHistory comments = new CommentHistory();
+
   @Override
-  public void init() {
-    BookRecs = new ArrayList<>();
-    BookRecs.add("The Thread by Victoria Hislop");
-    BookRecs.add("The Hunchback of Notre Dame by Victor Hugo");
-    BookRecs.add("The Brothers Karamazov by Fyodor Dostoevsky");
+  /** Load all comment history onload **/
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    String json = new Gson().toJson(comments);
+    System.out.println("got json string " + json);
+    response.getWriter().println(json);   
   }
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String book = BookRecs.get((int) (Math.random() * BookRecs.size()));
-    response.setContentType("text/html;");
-    response.getWriter().println(book);
+  /** Add comment submission to the comments object **/
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = getParameter(request, "text-input", "");
```

Comment:
All these raw strings, such as this and for the various routes should prolly be declared as constants in a Constants.java and imported.

Reply:
I'll make sure to fix that at next refactoring commit!